client.go

```diff
@@ -37,13 +42,23 @@ func Username() (string, error) {
 	return currentUser.Username, nil
 }
 
+// If the kerberos client is set, returns its principal.
+// Otherwise, defers to Username()
+func username(krb5Client *krb.Client) (string, error) {
```

Comment:
Call this `krbUsername` (and the comment should be something like `krbUsername returns the principle of krbClient, or the result of calling Username if krbClient is nil`)

Reply:
But also rather than the passthrough, I'd just have a branching if statement below